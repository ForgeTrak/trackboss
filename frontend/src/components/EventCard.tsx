import React from 'react';
import { Box, Center, Text, Heading, VStack, HStack, IconButton, Divider } from '@chakra-ui/react';
import { GoKebabVertical } from 'react-icons/go';
import { useNavigate } from 'react-router-dom';

interface cardProps {
    date: string,
    startTime: string,
    endTime: string,
    desc: string,
}

export default function EventCard(props: cardProps) {
    const navigate = useNavigate();
    const navigateToCalendar = () => {
        const path = '/calendar';
        navigate(path);
    };

    return (
        <Box
            onClick={navigateToCalendar}
            bg="white"
            boxShadow="md"
            border="1px"
            borderColor="gray.200"
            p={3}
            height={333}
        >
            <Center>
                <VStack p={0}>
                    <HStack>
                        <IconButton
                            aria-label="Calendar"
                            size="xs"
                            variant="ghost"
                            p="0"
                            m="0"
                            fontSize="24px"
                            color="gray.200"
                            icon={<GoKebabVertical />}
                        />
                        <Heading>Next Event</Heading>
                    </HStack>
                    <Divider />
                    <Text color="orange" fontSize="3xl">{props.date}</Text>
                    <Text fontSize="2xl">
                        {props.startTime}
                        -
                        {props.endTime}
                    </Text>
                    <Text fontSize="1xl">{props.desc}</Text>
                </VStack>
            </Center>

        </Box>
    );
}
