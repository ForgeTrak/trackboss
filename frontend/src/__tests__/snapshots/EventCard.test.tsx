import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import EventCard from '../../components/EventCard';
import '@testing-library/jest-dom/extend-expect';

describe('event card', () => {
    it('renders all props correctly', () => {
        const header = renderer.create(
            <BrowserRouter>
                <EventCard date="Dec 19th" name="pick rocks, kid" />
            </BrowserRouter>,
        ).toJSON();
        expect(header).toMatchSnapshot();
    });
});
