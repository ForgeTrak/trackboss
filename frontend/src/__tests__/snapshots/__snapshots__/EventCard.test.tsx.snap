// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all props correctly 1`] = `
<div
  className="css-1ovjvli"
  onClick={[Function]}
>
  <div
    className="css-gmuwbf"
  >
    <div
      className="chakra-stack css-1hkoj8r"
    >
      <div
        className="chakra-stack css-84zodg"
      >
        <button
          aria-label="Calendar"
          className="chakra-button css-16xtio7"
          type="button"
        >
          <svg
            aria-hidden={true}
            fill="currentColor"
            focusable={false}
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 3 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 2.5a1.5 1.5 0 1 0 3 0 1.5 1.5 0 0 0-3 0zm0 5a1.5 1.5 0 1 0 3 0 1.5 1.5 0 0 0-3 0zM1.5 14a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
              fillRule="evenodd"
            />
          </svg>
        </button>
        <h2
          className="chakra-heading css-0"
        >
          Next Event
        </h2>
      </div>
      <hr
        aria-orientation="horizontal"
        className="chakra-divider css-1upb9tn"
      />
      <p
        className="chakra-text css-1ez5l2u"
      >
        Dec 19th
      </p>
      <p
        className="chakra-text css-1n49pl7"
      >
        12:00pm
        -
        5:00pm
      </p>
      <p
        className="chakra-text css-cewpam"
      >
        pick rocks, kid
      </p>
    </div>
  </div>
</div>
`;
